lib/interfaces/include/BMSDriverGroup.h

```diff
@@ -0,0 +1,332 @@
+ * Each instance LTC6811 will represent functionality for all ICs on a particular chip select
+ * Therefore, the data stored in one instance will only hold HALF. One for chip_select 9 and the other for 10
+ * During instantiation, we need to make sure that the addresses are assigned for each LTC6811
+ */
+
+/* Library Includes */
+/**
+ * INCLUDE: LTC6811-1.h and LTC6811-2.h for model-specific data packaging
+ * LTC6811-1 can only function through broadcasted commands and therefore used daisy-chained
+ * LTC6811-2 can operate under broadcast and address functionality, but will only perform address-based commands
+ * Irregardless of which model we choose to use, we need will need access to Arduino and SPI repositories.
+ */
+#include <Arduino.h>
+#include <SPI.h>
+#include <string.h>
+#include <stdio.h>
+#include "Configuration.h"
+#include "LTCSPIInterface.h"
+
+class BMSDriverGroup
+{
+public:
+    BMSDriverGroup() = default;
+    BMSDriverGroup(int cs, int ic_num) : chip_select(cs),
+                                         ic_count(ic_num) {};
+
+    /* -------------------- SETUP FUNCTIONS -------------------- */
+
+    /**
+     * INIT: ONLY CALLED ONCE at initialization. LTC6811 only needs to set up the PEC table
+     * @post We should have initialized the PEC by calling generate_PEC_table(), so every other time
+     * we need to find the PEC it will be much faster.
+     */
+    void init();
+
+    /**
+     * PEC:
+     * The Packet Error Code (PEC) is a Error Checker–like CRC for CAN–to make sure that command and data
+     * bytes are communicated to the device correctly.
+     * This code was taken directly from the data sheet, page 76
+     * @post Instantiate the public pec15Table array of size 256. This variable should never be modified afterwards
+     */
+    void generate_PEC_table();
+
+    /**
+     * WAKEUP:
+     * The Wakeup protocol will move the device from the SLEEP to STANDBY / REFUP state
+     * Reference pages 51-52 on the data sheet for specific information.
+     * @pre we don't care; for us, it's good to perform a wakeup to guarantee proper data propogation
+     * @post essentially sets connected pin to LOW for period of time, then to HIGH for period of time
+     */
+    void start_wakeup_protocol();
+
+    /**
+     * Returns the delay time depending on what state we are in: SLEEP, IDLE, or MEASURE
+     * @return int number of milliseconds
+     */
+    int get_delay_time();
+
+    /* -------------------- READING DATA FUNCTIONS -------------------- */
+
+    /**
+     * Reads voltage data from ALL 4 Cell Voltage Registers on the device.
+     * A register, for those that don't know (I didn't), is memory.
+     * @pre in order to actually "read" anything, we need to call wakeup() and send data over SPI
+     * @post store all 126 cell voltages into the cell_voltages container
+     * AND record the max / minimum values and locations of them
+     */
+    void read_voltages();
```

Comment:
this should return the voltages. no need to store the voltages as member variables within this class instance.